packages/lesswrong/components/hooks/useForeignApolloClient.tsx

```diff
@@ -0,0 +1,6 @@
+import { createContext, useContext } from "react";
+import type { ApolloClient, NormalizedCacheObject } from '@apollo/client';
+
+export const ForeignApolloClientContext = createContext<ApolloClient<NormalizedCacheObject> | undefined>(undefined);
+
+export const useForeignApolloClient = () => useContext(ForeignApolloClientContext);
```

Comment:
Flagging that I don't know if this should be done in a context or not. I'll think more about it talking with you.